### DISCUSSION TITLE
[TCP] Can not get message from other client.

### DISCUSSION THREAD

**Discussion by VincentZe** [alec1o/Netly#24] [Q&A (Question and Answer)] [answered]
Hi, I encountered some problem when using this lib. I start a server, it can receive message from a client wrote by Netly. But when I test it with (cmd) telnet 10.10.3.21 12345 (or socket debugger), it successfully connected to the server. but got no message and close. I can confirm net package contains data. (by using WireShark) Is there anything wrong?
<img width="579" alt="Snipaste_2023-11-28_16-14-45" src="https://github.com/alec1o/Netly/assets/25062026/63c595c0-b840-449b-84f9-e53884813c02">

```
// server setup
public class Server : MonoBehaviour
{
    private TcpServer server;
    private Host host;
    void Start()
    {
        server = new TcpServer(framing: true);
        host =new Host("10.10.3.21", 12345);
        server.OnOpen(() =>
        {
            Debug.Log($"[Sever] started! {host}");
        });

        server.OnClose(() =>
        {
            Debug.Log($"[Sever] closed! {host}");
        });

        server.OnError((Exception e) => {
            // called when connection not opened
            Debug.Log($"[Sever] Connection error: {e}");
        });


        server.OnEnter((TcpClient client) => {
            Debug.Log($"[Sever] Client Enter ({client.Host})");

        });

        server.OnExit((TcpClient client) => {
            Debug.Log($"[Sever] Client Exit ({client.Host})");
        });


        server.OnData((TcpClient client,byte[] data) => {
            Debug.Log($"[Sever] Client Recieve ({client.Host}) : {NE.GetString(data)}");

            GenObject(data);
        });


        server.OnEvent((TcpClient client,string name, byte[] data) =>
        {
            Debug.Log($"[Sever] Client event ({name}) -> {NE.GetString(data)}");
        });

        const int backlog = 10;
        server.Open(host, backlog);
    }
}

```

```
//receive successed by server
void TaskWithParameters() {

    client = new TcpClient();
    host = new Host("10.10.3.21", 12345);
    client.Open(host);

    JSON js = new JSON();
    js.Add("typ", "Wall");
    js.Add("pos", new float[] {1.0f,2.5f,1.5f});
    js.Add("rot", new float[] {0f,0f,90.0f});
    js.Add("siz", new float[] {1.0f,1.12f,2.5f});

    string jsonstring=js.CreateString();

    //client.ToData(jsonstring);
    client.ToEvent("ping", jsonstring);
    Debug.Log("message sent");
    client.Close();
}
```
connected to the server but got no message:
`telnet 10.10.3.21 12345 some_message`
`nc 10.10.3.21 12345 some_message`
socket debugger message

**Comment by alec1o** [comment, marked as answer]
## Well. if you want use ``Netly`` with other library you must do:

### Option 1

```csharp
var server = new TcpServer(framing: false);
```

-  Disable Messsage Framing: Because ``Nelty`` uses ``custom message framing`` for ``performance`` and ``data security`` purposes so that a message is received in the same way it is sent. ``Message Framing`` is something that each library adds differently and netly also uses a different protocol than the other ``TCP libraries``.

> By disabling Message Framing you will use ``standard TCP`` (without framing), [read more about that](https://bit.ly/message-framing).

### Option 2

> You can also integrate the netly protocol into your client. [read more about that](https://netly.docs.kezero.com/#/framing?id=what-is-netly-message-framing-api)

### Notice:
Remember that if your server has message framing enabled, the client must also have it enabled, or the client connection will be disconnected.
If your client has the Framing Enabled message and your server is disabled, the connection will be disconnected. ``Because the party that has Message Framing enabled cannot analyze the message sent by the party that has it disabled.``

### NOTE:
Disablilita ``MessageFraming`` then send feedback about it. I believe will have other developers with the same problem and this will serve as a guide for them.
 
See Source Code:
[See part that disconnect when error on parse message](https://github.com/alec1o/Netly/blob/main/src/Tcp/TcpClient.cs#L222-L229)

**Reply by VincentZe** [reply]
Thank you! This solves my problem!

**Reply by vanhaodev** [reply]
is it ok if I use "framing = false"?
